src/components/Tree/expandCollapseButton.js

```diff
@@ -11,16 +12,27 @@ function getIcon(isExpanded) {
     return <RightArrow />;
 }
 
+function getButton(isExpanded, isLoading, onClick) {
+    const spinner = <BrandSpinner variant="brand" size="x-small" />;
+    return !isExpanded && isLoading ? (
+        <Button size="x-small" variant="base" icon={spinner} disabled />
+    ) : (
+        <Button size="x-small" icon={getIcon(isExpanded)} onClick={onClick} />
+    );
+}
+
 export default function ExpandCollapseButton(props) {
-    const { hasChildren, isExpanded, onClick } = props;
+    const { hasChildren, isExpanded, isLoading, onClick } = props;
+
     if (hasChildren) {
-        return <Button size="x-small" icon={getIcon(isExpanded)} onClick={onClick} />;
+        return getButton(isExpanded, isLoading, onClick);
```

Comment:
if this is a function then it could be a component, I don't think we need this extra function ... all the logic could be here

```js
if (isLoading) {
    return <Spinner />;
}
if (hasChildren) {
    <Button size="x-small" icon={getIcon(isExpanded)} onClick={onClick} />;
}
return null;
```

Reply:
received, working on it

Reply:
solved according to the comments